File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-correlation-swaps-5-4/CorrelationAmount.cpp
// CorrelationAmount.cpp 
#include "CorrelationAmount.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CorrelationAmount::CorrelationAmount(TiXmlNode* xmlNode)
: CalculatedAmount(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //correlationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* correlationNode = xmlNode->FirstChildElement("correlation");

   if(correlationNode){correlationIsNull_ = false;}
   else{correlationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- correlationNode , address : " << correlationNode << std::endl;
   #endif
   if(correlationNode)
   {
      if(correlationNode->Attribute("href") || correlationNode->Attribute("id"))
      {
          if(correlationNode->Attribute("id"))
          {
             correlationIDRef_ = correlationNode->Attribute("id");
             correlation_ = boost::shared_ptr<Correlation>(new Correlation(correlationNode));
             correlation_->setID(correlationIDRef_);
             IDManager::instance().setID(correlationIDRef_,correlation_);
          }
          else if(correlationNode->Attribute("href")) { correlationIDRef_ = correlationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { correlation_ = boost::shared_ptr<Correlation>(new Correlation(correlationNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Correlation> CorrelationAmount::getCorrelation()
{
   if(!this->correlationIsNull_){
        if(correlationIDRef_ != ""){
             return boost::shared_static_cast<Correlation>(IDManager::instance().getID(correlationIDRef_));
        }else{
             return this->correlation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Correlation>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-correlation-swaps-5-4/CorrelationAmount.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CorrelationAmount : CalculatedAmount
    {
        public CorrelationAmount(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode correlationNode = xmlNode.SelectSingleNode("correlation");
            
            if (correlationNode != null)
            {
                if (correlationNode.Attributes["href"] != null || correlationNode.Attributes["id"] != null) 
                {
                    if (correlationNode.Attributes["id"] != null) 
                    {
                        correlationIDRef_ = correlationNode.Attributes["id"].Value;
                        Correlation ob = new Correlation(correlationNode);
                        IDManager.SetID(correlationIDRef_, ob);
                    }
                    else if (correlationNode.Attributes["href"] != null)
                    {
                        correlationIDRef_ = correlationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        correlation_ = new Correlation(correlationNode);
                    }
                }
                else
                {
                    correlation_ = new Correlation(correlationNode);
                }
            }
            
        
        }
        
    
        #region Correlation_
        private Correlation correlation_;
        public Correlation Correlation_
        {
            get
            {
                if (this.correlation_ != null)
                {
                    return this.correlation_; 
                }
                else if (this.correlationIDRef_ != null)
                {
                    correlation_ = IDManager.getID(correlationIDRef_) as Correlation;
                    return this.correlation_; 
                }
                else
                {
                      return this.correlation_; 
                }
            }
            set
            {
                if (this.correlation_ != value)
                {
                    this.correlation_ = value;
                }
            }
        }
        #endregion
        
        public string correlationIDRef_ { get; set; }
        
    
        
    
    }
    
}

